### ISSUE TITLE
'Resources/LaunchScreen.storyboard' conflicts with 'Resources/LaunchScreen.storyboard' with random builds

### ISSUE THREAD

**Issue by anotherlab** [xamarin/xamarin-macios#6770]
### Steps to Reproduce

1. Create Xamarin.Forms solution using the Shell template
2. Compile the iOS app
. 

### Expected Behavior
App compiles

### Actual Behavior
Randomly the build fails with the following error:
The file 'Resources/LaunchScreen.storyboard' conflicts with 'Resources/LaunchScreen.storyboard'.

Closing the solution, deleting the bin & obj folders, and reopening the solution will allow the app to compile again.  About 10% to 20% the storyboard error will come back.  The project file has only one reference to the LaunchScreen.storyboard.

### Environment

```
Microsoft Visual Studio Enterprise 2019
Version 16.2.2
VisualStudio.16.Release/16.2.2+29209.62
Microsoft .NET Framework
Version 4.7.03056

Installed Version: Enterprise

Visual C++ 2019   00435-60000-00000-AA098
Microsoft Visual C++ 2019

Add New File   3.5.134
The fastest and easiest way to add new files to any project - including files that start with a dot

ADL Tools Service Provider   1.0
This package contains services used by Data Lake tools

ASP.NET and Web Tools 2019   16.2.290.48383
ASP.NET and Web Tools 2019

ASP.NET Web Frameworks and Tools 2019   16.2.290.48383
For additional information, visit https://www.asp.net/

Azure App Service Tools v3.0.0   16.2.290.48383
Azure App Service Tools v3.0.0

Azure Data Lake Node   1.0
This package contains the Data Lake integration nodes for Server Explorer.

Azure Data Lake Tools for Visual Studio   2.3.9000.0
Microsoft Azure Data Lake Tools for Visual Studio

Azure Functions and Web Jobs Tools   16.2.290.48383
Azure Functions and Web Jobs Tools

Azure Stream Analytics Tools for Visual Studio   2.3.9000.0
Microsoft Azure Stream Analytics Tools for Visual Studio

Bundler & Minifier   2.8.396
Adds support for bundling and minifying JavaScript, CSS and HTML files in any project.

C# Tools   3.2.0-beta4-19380-04+5e176ad8aea2724fe6787495c6fcc9144c550de1
C# components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Common Azure Tools   1.10
Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.

Cookiecutter   16.2.19178.1
Provides tools for finding, instantiating and customizing templates in cookiecutter format.

CSS Tools 2019   1.1.2
Provides additional features to the CSS editor in Visual Studio.

EditorConfig Language Service   1.17.260
Language service for .editorconfig files.

EditorConfig helps developers define and maintain consistent coding styles between different editors and IDEs.

Extensibility Message Bus   1.2.0 (d16-2@8b56e20)
Provides common messaging-based MEF services for loosely coupled Visual Studio extension components communication and integration.

Fabric.DiagnosticEvents   1.0
Fabric Diagnostic Events

File Icons   2.7
Adds icons for files that are not recognized by Solution Explorer

File Nesting   2.6.85
Automatically nest files based on file name and enables developers to nest and unnest any file manually

Image Optimizer   4.0.132
Uses industry standard tools to optimize any JPEG, PNG and Gifs - including animated Gifs. Can do both lossy and lossless optimization.

Image Sprites   1.4.58
Boost your website's performance by creating image sprites to reduce the amount of HTTP requests needed.

IntelliCode Extension   1.0
IntelliCode Visual Studio Extension Detailed Info

Markdown Editor   1.12.236
A full featured Markdown editor with live preview and syntax highlighting. Supports GitHub flavored Markdown.

Microsoft Azure HDInsight Azure Node   2.3.9000.0
HDInsight Node under Azure Node

Microsoft Azure Hive Query Language Service   2.3.9000.0
Language service for Hive query

Microsoft Azure Service Fabric Tools for Visual Studio   2.5
Microsoft Azure Service Fabric Tools for Visual Studio

Microsoft Azure Stream Analytics Language Service   2.3.9000.0
Language service for Azure Stream Analytics

Microsoft Azure Stream Analytics Node   1.0
Azure Stream Analytics Node under Azure Node

Microsoft Azure Tools   2.9
Microsoft Azure Tools for Microsoft Visual Studio 0x10 - v2.9.20626.2

Microsoft Continuous Delivery Tools for Visual Studio   0.4
Simplifying the configuration of Azure DevOps pipelines from within the Visual Studio IDE.

Microsoft JVM Debugger   1.0
Provides support for connecting the Visual Studio debugger to JDWP compatible Java Virtual Machines

Microsoft Library Manager   1.0
Install client-side libraries easily to any web project

Microsoft MI-Based Debugger   1.0
Provides support for connecting Visual Studio to MI compatible debuggers

Microsoft Visual C++ Wizards   1.0
Microsoft Visual C++ Wizards

Microsoft Visual Studio Tools for Containers   1.1
Develop, run, validate your ASP.NET Core applications in the target environment. F5 your application directly into a container with debugging, or CTRL + F5 to edit & refresh your app without having to rebuild the container.

Microsoft Visual Studio VC Package   1.0
Microsoft Visual Studio VC Package

Mono Debugging for Visual Studio   16.2.6 (4cfc7c3)
Support for debugging Mono processes with Visual Studio.

Multilingual App Toolkit   4.0
Multilingual App Toolkit helps you localize your Windows Store app by providing file management, pseudo and machine translation, translation editor, and build integration. http://aka.ms/matinstall

Node.js Tools   1.5.10610.1 Commit Hash:529a87de2769e143655e31ea81795c42e5775ef8
Adds support for developing and debugging Node.js apps in Visual Studio

NuGet Package Manager   5.2.0
NuGet Package Manager in Visual Studio. For more information about NuGet, visit https://docs.nuget.org/

Open Command Line   2.1.218
Opens a command line at the root of the project. Support for all consoles such as CMD, PowerShell, Bash etc. Provides syntax highlighting, Intellisense and execution of .cmd and .bat files.

Open in Visual Studio Code   1.4.46
Adds a menu command that lets you open any solution, project, folder and file in Visual Studio Code.

Package Installer   2.0.103
Makes it easier, faster and more convenient than ever to install Bower, npm, Yarn, JSPM, TSD, Typings and NuGet packages to any project

Package Security Alerts   1.0.22
Identifies npm and Bower packages that contains known security vulnerabilities to ensure your project is always using the most secure package versions.

ProjectServicesPackage Extension   1.0
ProjectServicesPackage Visual Studio Extension Detailed Info

Python   16.2.19178.1
Provides IntelliSense, projects, templates, debugging, interactive windows, and other support for Python developers.

Python - Conda support   16.2.19178.1
Conda support for Python projects.

Python - Django support   16.2.19178.1
Provides templates and integration for the Django web framework.

Python - IronPython support   16.2.19178.1
Provides templates and integration for IronPython-based projects.

Python - Profiling support   16.2.19178.1
Profiling support for Python projects.

ResourcePackage Extension   1.0
ResourcePackage Visual Studio Extension Detailed Info

ResourcePackage Extension   1.0
ResourcePackage Visual Studio Extension Detailed Info

SettingsWindow Extension   1.0
SettingsWindow Visual Studio Extension Detailed Info

Snapshot Debugging Extension   1.0
Snapshot Debugging Visual Studio Extension Detailed Info

SQL Server Data Tools   16.0.61906.28070
Microsoft SQL Server Data Tools

SVG Viewer   1.0.8
Makes it much easier to work with SVG files by showing a live preview in the bottom-right corner of the code window

ToolWindowHostedEditor   1.0
Hosting json editor into a tool window

TypeScript Tools   16.0.10627.2001
TypeScript Tools for Microsoft Visual Studio

Visual Basic Tools   3.2.0-beta4-19380-04+5e176ad8aea2724fe6787495c6fcc9144c550de1
Visual Basic components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Visual F# Tools 10.4 for F# 4.6   16.2.0-beta.19321.1+a24d94ecf97d0d69d4fbe6b8b10cd1f97737fff4
Microsoft Visual F# Tools 10.4 for F# 4.6

Visual Studio Code Debug Adapter Host Package   1.0
Interop layer for hosting Visual Studio Code debug adapters in Visual Studio

Visual Studio Tools for Containers   1.0
Visual Studio Tools for Containers

Visual Studio Tools for Kubernetes   1.0
Visual Studio Tools for Kubernetes

VisualStudio.Mac   1.0
Mac Extension for Visual Studio

Web Compiler   1.12.394
Compiler for LESS, Sass and CoffeeScript files

Web Essentials 2019   0.5.7
The easiest way to set up Visual Studio for the ultimate web development experience.

Xamarin   16.2.0.91 (d16-2@2983aebeb)
Visual Studio extension to enable development for Xamarin.iOS and Xamarin.Android.

Xamarin Designer   16.2.0.375 (remotes/origin/d16-2@357d38ef4)
Visual Studio extension to enable Xamarin Designer tools in Visual Studio.

Xamarin Templates   16.3.117 (59a59e8)
Templates for building iOS, Android, and Windows apps with Xamarin and Xamarin.Forms.

Xamarin.Android SDK   9.4.1.0 (d16-2/ebd65c9)
Xamarin.Android Reference Assemblies and MSBuild support.
    Mono: mono/mono/2019-02@e6f5369c2d2
    Java.Interop: xamarin/java.interop/d16-2@d64ada5
    LibZipSharp: grendello/LibZipSharp/d16-2@caa0c74
    LibZip: nih-at/libzip/rel-1-5-1@b95cf3f
    ProGuard: xamarin/proguard/master@905836d
    SQLite: xamarin/sqlite/3.27.1@8212a2d
    Xamarin.Android Tools: xamarin/xamarin-android-tools/d16-2@6f6c969


Xamarin.iOS and Xamarin.Mac SDK   12.14.0.114 (c669116)
Xamarin.iOS and Xamarin.Mac Reference Assemblies and MSBuild support.

ZenCoding   1.2.12
Provides ZenCoding for the HTML Editor - full support for static HTML, Razor and WebForms.
```

```
=== Visual Studio Enterprise 2019 for Mac ===

Version 8.2.3 (build 16)
Installation UUID: 44133281-7be2-4b64-a840-e927bfd6eb10
	GTK+ 2.24.23 (Raleigh theme)
	Xamarin.Mac 5.14.0.85 (d16-2-fix-mac-updates / 86973979)

	Package version: 600000319

=== Mono Framework MDK ===

Runtime:
	Mono 6.0.0.319 (2019-02/52203786470) (64-bit)
	Package version: 600000319

=== NuGet ===

Version: 5.1.0.6013

=== .NET Core ===

Runtime: /usr/local/share/dotnet/dotnet
Runtime Versions:
	2.1.12
	2.1.11
SDK: /usr/local/share/dotnet/sdk/2.1.701/Sdks
SDK Versions:
	2.1.701
	2.1.700
MSBuild SDKs: /Library/Frameworks/Mono.framework/Versions/6.0.0/lib/mono/msbuild/Current/bin/Sdks

=== Xamarin.Profiler ===

Version: 1.6.11
Location: /Applications/Xamarin Profiler.app/Contents/MacOS/Xamarin Profiler

=== Updater ===

Version: 11

=== Apple Developer Tools ===

Xcode 10.3 (14492.2)
Build 10G8

=== Xamarin.Mac ===

Version: 5.14.0.114 (Visual Studio Enterprise)
Hash: c6691164
Branch: d16-2
Build date: 2019-07-29 15:54:06-0400

=== Xamarin.iOS ===

Version: 12.14.0.114 (Visual Studio Enterprise)
Hash: c6691164
Branch: d16-2
Build date: 2019-07-29 15:54:05-0400

=== Xamarin.Android ===

Version: 9.4.1.0 (Visual Studio Enterprise)
Commit: xamarin-android/d16-2/ebd65c9
Android SDK: /Users/chris.miller/Library/Developer/Xamarin/android-sdk-macosx
	Supported Android versions:
		None installed

SDK Tools Version: 26.1.1
SDK Platform Tools Version: 28.0.2
SDK Build Tools Version: 28.0.3

Build Information: 
Mono: mono/mono/2019-02@e6f5369c2d2
Java.Interop: xamarin/java.interop/d16-2@d64ada5
LibZipSharp: grendello/LibZipSharp/d16-2@caa0c74
LibZip: nih-at/libzip/rel-1-5-1@b95cf3f
ProGuard: xamarin/proguard/master@905836d
SQLite: xamarin/sqlite/3.27.1@8212a2d
Xamarin.Android Tools: xamarin/xamarin-android-tools/d16-2@6f6c969

=== Microsoft Mobile OpenJDK ===

Java SDK: /Users/chris.miller/Library/Developer/Xamarin/jdk/microsoft_dist_openjdk_1.8.0.25
1.8.0-25
Android Designer EPL code available here:
https://github.com/xamarin/AndroidDesigner.EPL

=== Android Device Manager ===

Version: 1.2.0.60
Hash: 686c817
Branch: remotes/origin/d16-2
Build date: 2019-08-07 15:49:58 UTC

=== Xamarin Designer ===

Version: 16.2.0.379
Hash: 3f0652957
Branch: remotes/origin/d16-2
Build date: 2019-08-01 17:32:50 UTC

=== Xamarin Inspector ===

Version: 1.4.3
Hash: db27525
Branch: 1.4-release
Build date: Mon, 09 Jul 2018 21:20:18 GMT
Client compatibility: 1

=== Build Information ===

Release ID: 802030016
Git revision: f438e4b3ac81dbc614da2f2d9ac3c70ea2e5c350
Build date: 2019-08-09 09:07:01+00
Build branch: release-8.2
Xamarin extensions: 67f026710e882e318fb0c90077b429be3dc0e688

=== Operating System ===

Mac OS X 10.14.5
Darwin 18.6.0 Darwin Kernel Version 18.6.0
    Thu Apr 25 23:16:27 PDT 2019
    root:xnu-4903.261.4~2/RELEASE_X86_64 x86_64


```

### Build Logs
```
Error occurred while restoring NuGet packages: An item with the same key has already been added.
1>------ Build started: Project: Ride360, Configuration: Debug Any CPU ------
1>Multilingual App Toolkit v4.0.6914.0
1>Multilingual App Toolkit build started.
1>MultilingualResources\Ride360.es-US.xlf : warning : 5 resource strings with existing translations were updated and applied to the XLIFF file.  The related translations are unchanged, but should be checked to ensure they are still valid for the updated source string.
1>MultilingualResources\Ride360.fr-CA.xlf : warning : 5 resource strings with existing translations were updated and applied to the XLIFF file.  The related translations are unchanged, but should be checked to ensure they are still valid for the updated source string.
1>Multilingual App Toolkit build completed successfully. (23 ms)
1>ViewModels\StudentViewModal.cs(73,50,73,65): warning CS0108: 'StudentViewModal.PropertyChanged' hides inherited member 'BaseViewModel.PropertyChanged'. Use the new keyword if hiding was intended.
1>Views\SubViews\RunCollectionContentView.xaml.cs(23,50,23,65): warning CS0108: 'RunCollectionContentView.PropertyChanged' hides inherited member 'BindableObject.PropertyChanged'. Use the new keyword if hiding was intended.
1>obj\Debug\netstandard2.0\Views\StudentDetailPage.xaml.g.cs(23,51,23,57): warning CS0108: 'StudentDetailPage.Layout' hides inherited member 'VisualElement.Layout(Rectangle)'. Use the new keyword if hiding was intended.
1>Views\StudentDetailPage.xaml.cs(128,22,128,26): warning CS0108: 'StudentDetailPage.Icon' hides inherited member 'Page.Icon'. Use the new keyword if hiding was intended.
1>Views\AppShell.xaml.cs(136,104,136,106): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread.
1>ViewModels\StudentsViewModel.cs(72,20,72,33): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread.
1>Views\SelectDistrictPageV2.xaml.cs(53,41,53,43): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread.
1>Views\SelectDistrictPageV2.xaml.cs(21,44,21,61): warning CS0169: The field 'SelectDistrictPageV2._visibleDistricts' is never used
1>Ride360 -> d:\dev\ride360\Ride360\Ride360\bin\Debug\netstandard2.0\Ride360.dll
1>Done building project "Ride360.csproj".
2>------ Build started: Project: Ride360.iOS, Configuration: Debug iPhoneSimulator ------
2>  Executing SayHello Task to establish a connection to a Mac Server. 
2>  						 Properties: SessionId=d15a61fd7e2ff2b73add2955766b052a, 
2>  							ServerPort=52572, 
2>  							ServerAddress=10.83.32.80, 
2>  							ServerUser=REDACTED 
2>  							ServerPassword=, 
2>  							SSHKey=, 
2>  							SSHPassPhrase=,
2>  							AppName=Ride360.iOS,
2>  							ContinueOnDisconnected=False
2>  Connecting to Mac server 10.83.32.80...
2>  Executing SayHello Task to establish a connection to a Mac Server. 
2>  						 Properties: SessionId=d15a61fd7e2ff2b73add2955766b052a, 
2>  							ServerPort=52572, 
2>  							ServerAddress=10.83.32.80, 
2>  							ServerUser=REDACTED, 
2>  							ServerPassword=, 
2>  							SSHKey=, 
2>  							SSHPassPhrase=,
2>  							AppName=Ride360.iOS,
2>  							ContinueOnDisconnected=False
2>  Connecting to Mac server 10.83.32.80...
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Xamarin\iOS\Xamarin.iOS.Common.targets(1341,3): error : The file 'Resources/LaunchScreen.storyboard' conflicts with 'Resources/LaunchScreen.storyboard'.
========== Build: 1 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========
========== Deploy: 0 succeeded, 0 failed, 0 skipped ==========
```
### Example Project (If Possible)

> VS bug [#1055616](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1055616)

**Comment by spouliot** [comment]
I was not able to reproduce this locally. Can you

1. try to build on a Mac (without Windows) to see if this happens (it might be Windows specific);

2. ZIP and attach your solution (maybe something configuration got broken)

Thanks!

**Comment by jemunk** [comment]
I also get this error

**Comment by DanielEnid** [comment]
I have the same error on an old project without shell.

**Comment by mackayn** [comment]
Happens all the time.

**Comment by PicOLinO** [comment]
Xamarin Native. Also has this error. Restart VS helps a lot

**Comment by wouterattoptal** [comment]
I'm also seeing this

**Comment by mrutter** [comment]
Same problem here after updating my solution from XF 3.6 to XF 4.2.

The problem happens on Visual Studio 2019 Enterprise, on Windows, when I run the debugger for the second time (Run VS, Debug, Ok, Stop debugger, Debug again, I get the error).

My workaround is to run "Clean Solution" (right click on the solution node, choose "Clean Solution") before starting the debugger. It works but it's still frustating.

Thank you.
-m

**Comment by ryanbuening** [comment]
Also seeing this in Xamarin Forms 4.2.

**Comment by maexsp** [comment]
Also have this problem often. Xamarin Forms 4.2 Win10  VS2019

**Comment by Pirate911** [comment]
I got the same error today and solved it by adding new LaunchScreen then deleting it , and make sure you don't have any value set in the following properties in info.plist , Main Interface , Launch Screen .

**Comment by longlostbro** [comment]
rebuild fixed it for me.

**Comment by Rob-Houweling** [comment]
Experiencing the same issue today, with XF 4.2.0.815419.
The following options did not solve it:
- Clean project (iOS)
- Clean solution
- removing obj/bin folders manually (in iOS project folder)
- Rebuilding project
- Rebuilding solution

The only thing working for me was to restart VS2019.
My environment info:
Microsoft Visual Studio Enterprise 2019
Version 16.3.0
VisualStudio.16.Release/16.3.0+29318.209
Microsoft .NET Framework
Version 4.8.03761

Installed Version: Enterprise

Visual C++ 2019   00433-90050-44434-AA714
Microsoft Visual C++ 2019

ADL Tools Service Provider   1.0
This package contains services used by Data Lake tools

Application Insights Tools for Visual Studio Package   9.1.00913.1
Application Insights Tools for Visual Studio

ASP.NET and Web Tools 2019   16.3.282.33872
ASP.NET and Web Tools 2019

ASP.NET Web Frameworks and Tools 2019   16.3.282.33872
For additional information, visit https://www.asp.net/

Azure App Service Tools v3.0.0   16.3.282.33872
Azure App Service Tools v3.0.0

Azure Data Lake Node   1.0
This package contains the Data Lake integration nodes for Server Explorer.

Azure Data Lake Tools for Visual Studio   2.3.9000.0
Microsoft Azure Data Lake Tools for Visual Studio

Azure Functions and Web Jobs Tools   16.3.282.33872
Azure Functions and Web Jobs Tools

Azure Stream Analytics Tools for Visual Studio   2.3.9000.0
Microsoft Azure Stream Analytics Tools for Visual Studio

C# Tools   3.3.1-beta3-19461-02+2fd12c210e22f7d6245805c60340f6a34af6875b
C# components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Common Azure Tools   1.10
Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.

Extensibility Message Bus   1.2.0 (d16-2@8b56e20)
Provides common messaging-based MEF services for loosely coupled Visual Studio extension components communication and integration.

Fabric.DiagnosticEvents   1.0
Fabric Diagnostic Events

IntelliCode Extension   1.0
IntelliCode Visual Studio Extension Detailed Info

Microsoft Azure HDInsight Azure Node   2.3.9000.0
HDInsight Node under Azure Node

Microsoft Azure Hive Query Language Service   2.3.9000.0
Language service for Hive query

Microsoft Azure Service Fabric Tools for Visual Studio   16.0
Microsoft Azure Service Fabric Tools for Visual Studio

Microsoft Azure Stream Analytics Language Service   2.3.9000.0
Language service for Azure Stream Analytics

Microsoft Azure Stream Analytics Node   1.0
Azure Stream Analytics Node under Azure Node

Microsoft Azure Tools   2.9
Microsoft Azure Tools for Microsoft Visual Studio 0x10 - v2.9.20816.1

Microsoft Continuous Delivery Tools for Visual Studio   0.4
Simplifying the configuration of Azure DevOps pipelines from within the Visual Studio IDE.

Microsoft JVM Debugger   1.0
Provides support for connecting the Visual Studio debugger to JDWP compatible Java Virtual Machines

Microsoft Library Manager   2.0.83+gbc8a4b23ec
Install client-side libraries easily to any web project

Microsoft MI-Based Debugger   1.0
Provides support for connecting Visual Studio to MI compatible debuggers

Microsoft Visual C++ Wizards   1.0
Microsoft Visual C++ Wizards

Microsoft Visual Studio Tools for Containers   1.1
Develop, run, validate your ASP.NET Core applications in the target environment. F5 your application directly into a container with debugging, or CTRL + F5 to edit & refresh your app without having to rebuild the container.

Microsoft Visual Studio VC Package   1.0
Microsoft Visual Studio VC Package

Mono Debugging for Visual Studio   16.3.7 (9d260c5)
Support for debugging Mono processes with Visual Studio.

NuGet Package Manager   5.3.0
NuGet Package Manager in Visual Studio. For more information about NuGet, visit https://docs.nuget.org/

ProjectServicesPackage Extension   1.0
ProjectServicesPackage Visual Studio Extension Detailed Info

Snapshot Debugging Extension   1.0
Snapshot Debugging Visual Studio Extension Detailed Info

SQL Server Data Tools   16.0.61908.27190
Microsoft SQL Server Data Tools

ToolWindowHostedEditor   1.0
Hosting json editor into a tool window

TypeScript Tools   16.0.10821.2002
TypeScript Tools for Microsoft Visual Studio

Visual Basic Tools   3.3.1-beta3-19461-02+2fd12c210e22f7d6245805c60340f6a34af6875b
Visual Basic components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Visual F# Tools 10.4 for F# 4.6   16.3.0-beta.19455.1+0422ff293bb2cc722fe5021b85ef50378a9af823
Microsoft Visual F# Tools 10.4 for F# 4.6

Visual Studio Code Debug Adapter Host Package   1.0
Interop layer for hosting Visual Studio Code debug adapters in Visual Studio

Visual Studio Tools for Containers   1.0
Visual Studio Tools for Containers

Visual Studio Tools for Kubernetes   1.0
Visual Studio Tools for Kubernetes

VisualStudio.Mac   1.0
Mac Extension for Visual Studio

Xamarin   16.3.0.274 (d16-3@06531f8)
Visual Studio extension to enable development for Xamarin.iOS and Xamarin.Android.

Xamarin Designer   16.3.0.230 (remotes/origin/d16-3-xcode11@bbe518670)
Visual Studio extension to enable Xamarin Designer tools in Visual Studio.

Xamarin Templates   16.3.565 (27e9746)
Templates for building iOS, Android, and Windows apps with Xamarin and Xamarin.Forms.

Xamarin.Android SDK   10.0.0.43 (d16-3/8af1ca8)
Xamarin.Android Reference Assemblies and MSBuild support.
    Mono: mono/mono/2019-06@7af64d1ebe9
    Java.Interop: xamarin/java.interop/d16-3@5836f58
    LibZipSharp: grendello/LibZipSharp/d16-3@71f4a94
    LibZip: nih-at/libzip/rel-1-5-1@b95cf3f
    ProGuard: xamarin/proguard/master@905836d
    SQLite: xamarin/sqlite/3.27.1@8212a2d
    Xamarin.Android Tools: xamarin/xamarin-android-tools/d16-3@cb41333


Xamarin.iOS and Xamarin.Mac SDK   13.2.0.42 (5e8a208)
Xamarin.iOS and Xamarin.Mac Reference Assemblies and MSBuild support.

**Comment by Catelinecz** [comment]
I have the same issue using VS2019 Community, W10. Clean doesn't work, neither does deleting cache on mac. Only restarting VS did the trick.

**Comment by rolfbjarne** [comment]
Please enable diagnostic build logs:
- On Visual Studio for Mac: Preferences > Projects > Build
- On Visual Studio for Windows: Tools > Options > Projects and Solutions > Build and Run
On Visual Studio for Windows you also want to add `-v -v -v -v` to the mtouch additional arguments by right-clicking the project in the solution explorer and selecting `Properties`.

and then when the build fails, attach the diagnostic build log here.

**Comment by REDECODE** [comment]
Same for me. 
I need to Unpair and than Pair to Mac and than the error disappear.

I use VS2019 with Xamarin.iOS and Xamarin.Mac SDK 13.4.0.2

**Comment by HannesGanthalerESec** [comment]
Same happens to me. Restarting VS helped.

**Comment by whitneyschmidt** [comment]
We have not received the requested information.

If you are still experiencing this issue please provide all the requested information then click the Reopen Issue button.

Thanks!

**Comment by kuntalkishore** [comment]
This shouldn't be closed. This is such an annoying issue, happens every time and unpairing and repairing works. However the downside is that after repairing the build again takes awfully longer time, very frustrating as it keeps happening on a regular basis.

I am using VS 2019 16.3 community version with Xam iOS 13.6.0.12 with Mac OS Catalina. This type of bug has been persistent with Xamarin Development since earlier versions of Visual Studio. With some release this goes away and again comes back with newer releases. Very frustrating and annoying.

**Comment by dotMorten** [comment]
I've been seeing this a lot lately on several different projects, new and old. A restart of VS and sometimes blowing away obj and bin folders resolves it.

**Comment by andrewleader** [comment]
I just saw this today on my iOS project for the first time. Open source project: https://github.com/powerplanner/powerplannerapps

**Comment by chamons** [comment]
Hey everyone 👋 

I understand the frustration, but as Whitney noted unless we have the information we need we can't move forward on pinning this down.

Ideally we need a project reproducing this and steps to reproduce. Now as it is random, unless someone figures out how to make it happen consistently we can't go down that route.

Because of that, we need the logs that Rolf and Whitney asked for - https://github.com/xamarin/xamarin-macios/issues/6770#issuecomment-541233363

Once we get diagnostic logs with the settings noted in that comment, we can reopen and investigate this issue.

Thanks everyone!

**Comment by dotMorten** [comment]
At this point it's pretty clear this is a genuine issue, and as such should be reopened so others will more easily find it. Closing bugs because you don't have all the info yet seems counter productive

**Comment by chamons** [comment]
@dotMorten - We had this bug open for **months** without anyone getting us a diagnostic build log. Without a full debug log, as we are unable to make any progress on it, as we can not reproduce it on our end.

By leaving it as closed, we are trying to clearly communicate that we can not make progress without people getting us the required info.

**Comment by kuntalkishore** [comment]
Honestly it really doesn't matters what project it is, it happens with each and every project. I could create a blank Xamarin project from template and it would still do this same thing.

Just to clarify, in my case, it's not just this particular error, sometimes the error would be codesigned exit with error code 1, the other time it would say something entirely different! All of these would go away after closing Visual studio, cleaning and rebuilding.

________________________________

**Comment by dotMorten** [comment]
@kuntalkishore if it is so easy to reproduce for you, could you provide the logs? It's only intermittent for me

**Comment by kuntalkishore** [comment]
Sure, if you can please provide the steps on how to grab the logs.


________________________________

**Comment by jimberg98** [comment]
This happens a lot to me and it happens with different things.  It's obvious that parameters are being duplicated somewhere for some reason.  The annoying workaround is to go to Tools/iOS/Pair to Mac to  disconnect and then reconnect.  This temporarily fixes the problem. 

What does disconnecting and reconnecting change?  Do files get cleaned up somehow during this process?  Does memory get cleaned up during the process?  That's the key to the problem.  An easy fix would be to have it disconnect and reconnect automatically when you go to build and run the application.

Is there supposed to be some sort of cleanup process when you stop the application that isn't getting completed?  A background task getting run that doesn't stop correctly?

I also have issues where I hit run and it builds, but doesn't actually run on the simulator.  Hitting run again works.  I think this may be related somehow.

I don't see a fix for this issue so why is it closed?  I'm using VS 2019 enterprise and using Xamarin Forms 4.4.

**Comment by chamons** [comment]
@jimberg98 See my [earlier comment](https://github.com/xamarin/xamarin-macios/issues/6770#issuecomment-568586912) on why this is closed.

I want to remind everyone that we need diagnostic logs of the issue to proceed. The instruction are here [here](https://docs.microsoft.com/en-us/xamarin/cross-platform/troubleshooting/questions/version-logs#verbose-build-output).

**Comment by bobwhitten** [comment]
Just updated to VS 16.4.3 and XCode 11.3.1, and I'm running into this as well.  Simple XF app on iOS. XF 4.4 - using Content Pages, not Shell.  Cleaning, deleting bin/obj and restarting VS seems to clear up.

I've enabled verbosity, so next time it shows up I'll capture the log and forward.

**Comment by jimberg98** [comment]
Here is a log where I get the error :
[Diagnostic Log For Error.log](https://github.com/xamarin/xamarin-macios/files/4078691/Diagnostic.Log.For.Error.log)

I redacted some personal info with symbols but I used different symbols for the 2 words I redacted so I wouldn't make things ambiguous.

When I turned the diagnostic log on, the error has occurred a lot less frequently.  My guess is that it's likely a timing issue related with multiple processes running.  The extra logging slows things down enough to allow process to sync up better.

Does this error occur for people developing directly on a Mac?  I should have mentioned in my first post that I am using WIndows 10 and pairing with a Mac for testing.  It would also be nice if someone answered the questions in my original post about what occurs when the PC pairs with the Mac.

Thanks.

**Comment by mikebikerider** [comment]
I have the same issue in VS 2019 16.4.3, Xamarin Forms 4.4.0.991537, NETStandard.Library 2.0.3 . XCode 11.3.1.
Disconnecting from MAC and connecting back usually helps for a run or two, and then it is back again. It started about 3 month ago and it is getting worse.
On occasion the LaunchScreen.storyboard output Error is followed by warnings about missing references to Nuget assemblies, and the References branch in the iOS project adds nodes with yellow marks. The iOS project file remains unchanged. Once VS2019 reloaded the extra references disappear and after build the app is deployed onto device until next time.

**Comment by jimberg98** [comment]
It occurs in 4.5.0.142-pre1, too.  @mikebikerider turn on diagnostic logging and see if slowing things down with all of the extra logging reduces the number of occurrences.  It has for me and makes it somewhat bearable to work with.  Under Tools/Options/Projects and Solutions/Build and Run.  I set both the MSBuild project build values to Diagnostic.

**Comment by bobwhitten** [comment]
FYI, I haven't had this problem since I turned on the extra logging.  When the lights go on, the bugs run.

**Comment by jimberg98** [comment]
@bobwhitten I still have it, but it's definitely less frequent.  You have to love Heisenbugs. :-)  I'm still trying to figure out what happens in the whole connect/disconnect process that cleans up the problem.  Whatever it is should be done as every build/run cycle.  That would be the easiest fix.

**Comment by mikebikerider** [comment]
@mikebikerider turn on diagnostic logging... extra logging . 
Device Log toolbar button?

**Comment by PavaniTatikonda27** [comment]
Please check in .csproj file (by right click on ios project -> edit Project File) if there are any duplicates for the launchScreen.storyboard. In my case there are duplicates once I removed them it resolves and able to run the project.

**Comment by mikebikerider** [comment]
Thanks for the suggestion, but no duplicates of the LaunchScreen.storyboard in the project file.
The issue went away after a couple of VS2019 updates released since January 2020. So far I had not a single occurrence of this nasty issue under VS 16.6.3. Xamarin.Forms  4.7 .

**Comment by andrewleader** [comment]
Yeah I don't think I've seen this issue recently either (I was doing some iOS development a couple weeks ago and didn't run into any build/deploy errors)
